src/utils/path_utils/src/path_utils.c

```diff
@@ -59,3 +60,82 @@ STRING_HANDLE SanitizePathSegment(const char* unsanitized)
 
     return sanitizedHandle;
 }
+
+/**
+ * @brief Concatenates a directory and folder to form a path.
+ * @param dirPath The directory path (eg /var/lib/adu)
+ * @param folderName The folder name (eg 12345678-1234-1234-1234-123456789012)
+ * @return char* The concatenated path, or NULL on error.
+*/
+char* PathUtils_ConcatenateDirAndFolderPaths(const char* dirPath, const char* folderName)
+{
+    char* ret = NULL;
+    char* tempRet = NULL;
+    char dir[PATH_MAX + 1] = { 0 }; // Plus one for null terminator
```

Comment:
PATH_MAX already includes for the null terminator, so should remove + 1 here: https://www.gnu.org/software/libc/manual/html_node/Limits-for-Files.html